src/ProjectManagement/Utility/BuildIntegratedProjectUtility.cs

```diff
@@ -79,5 +77,77 @@ public static ExternalProjectReference ConvertProjectReference(BuildIntegratedPr
         {
             return new ExternalProjectReference(reference.Name, reference.PackageSpecPath, reference.ExternalProjectReferences);
         }
+
+        public static IReadOnlyList<PackageIdentity> GetOrderedProjectDependencies(BuildIntegratedNuGetProject buildIntegratedProject)
```

Comment:
Fair enough. I'm OK with all this logic, just wondering if we can simplify it a bit and make it as declarative as possible :).